### PR TITLE
spirv-fuzz: Transformation to add a new function to a module

### PR DESCRIPTION
**Pull request by afd**
This adds a large transformation that can add a new function to a
SPIR-V module.  This paves the way for donation of code from one
module to another.

### CODE REVIEW

File: source/fuzz/transformation_add_function.h [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+  // Adds the function defined by |message_.instruction| to the module
+  void Apply(opt::IRContext* context, FactManager* fact_manager) const override;
+
+  protobufs::Transformation ToMessage() const override;
+
+ private:
+  // Attempts to create a function from the series of instructions in
+  // |message_.instruction| and add it to |context|.  Returns false if this is
+  // not possible due to the messages not respect the basic structure of a
```
Comment (paulthomson) [inline source/fuzz/transformation_add_function.h]: respect -> respecting

File: test/fuzz/transformation_add_function_test.cpp [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+
+  const auto env = SPV_ENV_UNIVERSAL_1_4;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  ASSERT_TRUE(IsValid(env, context.get()));
+
+  FactManager fact_manager;
+
+  TransformationAddFunction transformation1(std::vector<protobufs::Instruction>(
+      {MakeInstructionMessage(
```
Comment (paulthomson) [inline test/fuzz/transformation_add_function_test.cpp]: Up to you, but could use `// clang-format off` and `// clang-format on`. And then keep each message on its own line.

File: source/fuzz/transformation_add_function.h (hunk 1 above)
Comment (afd) [reply source/fuzz/transformation_add_function.h]: Fixed.

File: test/fuzz/transformation_add_function_test.cpp (hunk 2 above)
Comment (afd) [reply test/fuzz/transformation_add_function_test.cpp]: Thanks for the tip, though I'm OK with what clang-format has done here.

**Pull request merged**